bumble/l2cap.py

```diff
@@ -2095,7 +2155,34 @@ async def open_le_coc(
 
         return channel
 
-    async def connect(self, connection: Connection, psm: int) -> Channel:
+    async def connect(
```

Comment:
Having a single method here to dispatch to `_open_le_credit_based_channel` or `_open_classic_channel` depending on `mode` is a bit misleading, because not all parameters apply to all modes. Having a separate function for each mode is cleaner, as there are not extra/unused arguments. Also, calling `connect` for a connectionless channel is surprising.  Maybe `Device.connect_l2cap_le_credit_based_channel()` for LE COC, and `Device.open_l2cap_channel()` for non-connection-oriented channels?

Reply:
The problem here is that it makes function name very long, and will require a 2(for client/server) * 2(for device/l2cap) of public methods for each mode which is a little bit fat. Also, it is not an anti-pattern to ignore some parameters. For example, `asyncio.open_connection()` has SSL parameters but they will be ignored in pure TCP mode.

Reply:
In that case, would it be better to saperate `open_channel()` and `connect()`, where `open_channel()` resides in both `device` and `l2cap` only returning an object from a type, while `connect()` only resides in l2cap, taking mode-specific parameters?
Example:
```py
# device.py

def open_l2cap_channel(mode: ...) -> l2cap.Channel:
  ...

# l2cap.py
class ClassicBasicChannel:
  ...
  def connect(psm: int):
    ...

class LeCreditBasedChannel:
  ...
  def connect(psm: int, credit, mtu, mps):
    ...

def open_channel(mode: ...) -> l2cap.Channel:
  ...

```

Reply:
For server it will be a little bit complicated, because the API doesn't return any object.

Reply:
Or with a new proxy...
```py
class ClassicBasicChannelServer:
    def __init__(self, psm: int) -> None:
        self.psm = psm

    async def serve(self, handler: Callable[[Channel], Any]) -> None:
        ...


class LeCreditBasedChannelServer:
    def __init__(self, psm: int) -> None:
        self.psm = psm

    async def serve(
        self, handler: Callable[[LeConnectionOrientedChannel], Any], credit, mtu, mps
    ) -> None:
        ...

ServerMode = TypeVar('ServerMode', ClassicBasicChannelServer, LeCreditBasedChannelServer)

class ChannelManager:
    def open_server(mode: ServerMode) -> ServerMode:
        ...

```